### ISSUE TITLE
Consider offering a dual license MIT and CC0

### ISSUE THREAD

**Issue by bfattori** [reactive-streams/reactive-streams-jvm#434]
Is there any way you could dual license your project as MIT and CC0? The MIT is a fair license and allows for the broadest usage. I propose this because for businesses, the idea of the Public Domain isn't a legal framework that applies globally. Larger organizations cannot consume components licensed in this way because they present unique problems around the globe.

Allowing for the dual licensing of the project as either MIT or CC0 would satisfy you projects goals of being open an unencumbered by those who wish to use it that way, but still legally acceptable to businesses.

**Comment by viktorklang** [comment]
@bfattori Thanks for raising this question, Brett. My initial reaction is: Is this really a problem? I say this because this code is in the JDK: https://docs.oracle.com/javase/9/docs/api/java/util/concurrent/Flow.html and basically all of JSR166 is Public Domain: http://gee.cs.oswego.edu/dl/concurrency-interest/

Fundamentally, the TCK wouldn't be packaged with any application code, so it is a test-time dependency, so it wouldn't be a problem. I guess it could be an issue if you want to take the example code and paste verbatim?

Do you have any pointers to material which I could read up on w.r.t. this topic?

Thanks!
√

**Comment by bfattori** [comment]
@viktorklang I was unaware that reactive-streams is included in the JDK. You also say that it is a testing-time only component? If this is so, then it's not necessarily a problem.

Thanks for the quick response.

**Comment by viktorklang** [comment]
@bfattori The interfaces is in the JDK (as well), the TCK is a test-time component. What remains is the examples, and those are CC0. So I guess those would only pose a problem in your case if you included them in production code? (I am not a lawyer, doctor, etc)

**Comment by bfattori** [comment]
@viktorklang If we did start with the examples, that are licensed CC0, and modified them, then the code would need to remain CC0, in my understanding. That could pose a problem if the Public Domain is not easily recognized in another region. A dual license wouldn't change or hurt your examples in any way, but would offer companies a "safer" path to derive from the examples.

**Comment by bfattori** [comment]
Also, I suppose a good way to see how "Public Domain" is defined around the world (and why it's complicated) is to view the many different regional sites on Wikipedia regarding the Public Domain:

https://en.wikipedia.org/wiki/Public_domain

**Comment by akarnokd** [comment]
> open an unencumbered by those who wish to use it that way, but still legally acceptable to businesses.

Do you have a specific company in mind that has trouble with CC0 or is this a hypotetical scenario? 

[This FAQ](https://wiki.creativecommons.org/wiki/CC0_FAQ) indicates CC0 is pretty reasonable and unless your company wants to reinvent it, it should not be of a problem (see [this](https://www.slant.co/versus/3575/3574/~mit-license_vs_the-unlicense) short comparison). 

Most users don't interact with Reactive Streams directly but through other libraries, licensed typically under Apache 2.0, where [including CC0 is fine](https://www.apache.org/legal/resolved.html#can-works-placed-in-the-public-domain-be-included-in-apache-products). (Fun fact, [Doug Lea](https://www.apache.org/legal/resolved.html#concurrent) comes up too.)

In summary, I don't consider this to be a problem and recommend **not** dual licensing Reactive Streams, which I've contributed to as well under the CC0 premise.

**Comment by bfattori** [comment]
I have a sector of companies that might consider this a problem: _those that are regulated by government organizations, and not just by the U.S. government, in their field of endeavour._ 

Because the Public Domain isn't a consistent legal framework, a company that is using the software must comply with the license terms as best they can. In a country where the Public Domain does not exist, and the fallback to a "free license" doesn't satisfy, that would limit the usability of the component. Just because it is used with the Apache 2.0, does not change its terms though. The Apache component uses the CC0 component, with the two licenses being compatible in the U.S.

If a company does work abroad, and they are regulated, their could be a potential problem with using the CC0 license. It hasn't been tested and just assuming it's okay won't fly when there is a multi-million dollar budget resting on its merits.

**Comment by viktorklang** [comment]
Disclaimer: I am not a lawyer, expert on IP legislation, etc. YMMV. 

@bfattori So given that the interfaces is in the JDK and under the JDK license, the TCK is a test-time dependency your remaining concern is if someone possibly derives code from the examples-jar?
Dual-licensing the examples seems—to me—to be unnecessary for end-users since end-users should be using an RS implementation (such as Akka Streams, Reactor, RxJava etc) and for RS implementors the examples only serve as educational material rather than a basis for their own implementation work.

@reactive-streams/contributors Anyone have any thoughts here?

**Comment by DougLea** [comment]
I agree with Viktor that there is no issue here. The interfaces are available under OpenJDK license as well as the original CC0. (Not especially relevant, but similarly for the OpenJDK TCK that primarily tests SubmissionPublisher).

**Comment by viktorklang** [comment]
Thanks @bfattori, I'll close this issue.
